pandas/_libs/groupby.pyx

```diff
@@ -388,10 +388,10 @@ def group_fillna_indexer(ndarray[int64_t] out, ndarray[intp_t] labels,
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def group_any_all(int8_t[::1] out,
-                  const int8_t[::1] values,
-                  const intp_t[:] labels,
-                  const uint8_t[::1] mask,
+def group_any_all(int8_t[:, ::1] out,
+                  const int8_t[:, :] values,
+                  const intp_t[::1] labels,
+                  const uint8_t[:, :] mask,
```

Comment:
See question below about if `mask` can be `None`, maybe it would be too verbose but could start using the `not None` addition to function definitions to make these explicit (for example `const uint8_t[:, :] mask not None` will reject `None` inputs